src/gentropy/dataset/l2g_prediction.py

```diff
@@ -83,3 +85,36 @@ def from_credible_set(
         )
 
         return l2g_model.predict(fm, session)
+
+    def to_disease_target_evidence(
+        self: L2GPrediction,
+        study_locus: StudyLocus,
+        study_index: StudyIndex,
+        l2g_threshold: float = 0.05,
+    ) -> DataFrame:
+        """Convert locus to gene predictions to disease target evidence.
+
+        Args:
+            study_locus (StudyLocus): Study locus dataset
+            study_index (StudyIndex): Study index dataset
+            l2g_threshold (float): Threshold to consider a gene as a target. Defaults to 0.05.
+
+        Returns:
+            DataFrame: Disease target evidence
+        """
+        datasource_id = "gwas_credible_sets"
+        datatype_id = "genetic_association"
+
+        return (
+            self.df.filter(f.col("score") >= l2g_threshold)
+            .join(study_locus.df, on="studyLocusId", how="inner")
```

Comment:
To make the join lighter
```suggestion
            .join(study_locus.df.select("studyLocusId", "studyId"), on="studyLocusId", how="inner")
```

Reply:
I suspect spark optimizes this in the background, but it's better to be more explicit.